.github/actions/aws-configure-cli/action.yml

```diff
@@ -0,0 +1,45 @@
+---
+name: Configure AWS CLI
+description: Import AWS Secrets from Vault and configure AWS CLI profile
+# The official action do not support profiles
+
+inputs:
+    vault-addr:
+        description: The URL of the Vault server
+        required: true
+    vault-role-id:
+        description: The Vault Role ID
+        required: true
+    vault-secret-id:
+        description: The Vault Secret ID
+        required: true
```

Comment:
personal preference to simplify it by just passing in the secrets but this approach also works.